session/storage.go

```diff
@@ -0,0 +1,135 @@
+package session
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/influxdata/influxdb/v2"
+)
+
+type Store interface {
```

Comment:
After reflecting on some of my comments, I think this interface is the problem here. We have a Storage type and a Store interface. This Store interface has come session behaviour semantics mixed with key value store requirements.

I personally would just do away with this interface. Instead, I would have a Storage interface defined next to the Service which the concrete type here satisfies and have a few implementations of that interface. e.g. an `inmem.SessionStorage` and a `bolt.SessionStorage`.

Reply:
As per our conversation this has been resolved by keeping the store interface and removing the storage interface inside session. I believe we would like to draw the line between the data storage layer and the session service. By placing the interface on the store we can translate session to keys in a specific way and allow the back end storage handle expiration.